docs/guides/grafana-vmgateway-openid-configuration.md

```diff
@@ -0,0 +1,162 @@
+Those tokens can include information about the user and the tenant they belong to, which can be used
+to restrict access to metrics to only those that belong to the tenant.
+
+## Prerequisites
+
+* Identity service that can issue JWT tokens
+* Grafana
+* VMSingle or VMCluster
+* vmgateway
+
+## Configure identity service
+
+The identity service must be able to issue JWT tokens with the following `vm_access` claim. Here is an example claims
+object:
+
+```json
+{
+  "vm_access": {
+    "tenant_id": {
+      "account_id": 0,
+      "project_id": 0
+    }
+  }
+}
+```
+
+You can find details about all supported options in the
+vmgateway [documentation](https://docs.victoriametrics.com/vmgateway.html#access-control).
+
+### Configuration for Keycloak
+
+1. Log in with admin credentials to your Keycloak instance
+2. Go to `Clients` -> `Create`.
+   Use `OpenID Connect` as `Client Type`.
+   Specify `grafana` as `Client ID`.
+   Click `Next`.
+   <img src="grafana-vmgateway-openid-configuration/create-client-1.png" width="800">
+3. Enable `Client authentication`.
+   Enable `Authorization`.
+   <img src="grafana-vmgateway-openid-configuration/create-client-2.png" width="800">
+   Click `Next`.
+4. Add Grafana URL as `Valid Redirect URIs`. For example, `http://localhost:3000/`.
+   <img src="grafana-vmgateway-openid-configuration/create-client-3.png" width="800">
+   Click `Save`.
+5. Go to `Clients` -> `grafana` -> `Credentials`.
+   <img src="grafana-vmgateway-openid-configuration/client-secret.png" width="800">
+   Copy the value of `Client secret`. It will be used later in Grafana configuration.
+6. Go to `Clients` -> `grafana` -> `Client scopes`.
+   Click at `grafana-dedicated` -> `Add mapper`.
+   <img src="grafana-vmgateway-openid-configuration/create-mapper-1.png" width="800">
+   <img src="grafana-vmgateway-openid-configuration/create-mapper-2.png" width="800">
+   Configure the mapper as follows
+    - `Mapper Type` as `User Attribute`.
+    - `Name` as `vm_access`.
+    - `Token Claim Name` as `vm_access`.
+    - `User Attribute` as `vm_access`.
+    - `Claim JSON Type` as `JSON`.
+      Enable `Add to ID token` and `Add to access token`.
+      <img src="grafana-vmgateway-openid-configuration/create-mapper-3.png" width="800">
+      Click `Save`.
+7. Go to `Users` -> select user to configure claims -> `Attributes`.
+   Specify `vm_access` as `Key`.
+   Specify `{"tenant_id" : {"account_id": 0, "project_id": 0 }}` as `Value`.
+   <img src="grafana-vmgateway-openid-configuration/user-attributes.png" width="800">
+   Click `Save`.
+
+## Configure grafana
+
+It is required to configure Grafana to use OpenID Connect authentication and to forward the JWT token to vmgateway.
+
+Example configuration for Grafana OpenID Connect authentication:
+```ini
+[auth.generic_oauth]
+enabled = true
+allow_sign_up = true
+team_ids =
+allowed_organizations =
```

Comment:
are these required to be listed with empty values?

Reply:
Nope, removed those